themes/bootstrap3/templates/RecordTab/overdriveholdings.phtml

```diff
@@ -50,21 +52,27 @@
         <?php endif; ?>
       </p>
 
-      <?php if (!$checkedOut || $checkedOutResult->isMagazine): ?>
+      <?php if ((!$checkedOut || $checkedOutResult->isMagazine) && ($supportsPatronActions || count($urls))): ?>
         <?php if (($avail->data->copiesAvailable ?? 0) > 0): ?>
           <p>
             <a class="btn btn-primary checkout" data-lightbox title="<?=$this->transEsc('request_place_text')?>"
-              href="<?="$hold_url?od_id=" . urlencode($od_id) . '&rec_id=' . urlencode($rec_id) . '&action=checkoutConfirm'?>">
+              href="<?=$supportsPatronActions ?
+                "$hold_url?od_id=" . urlencode($od_id) . '&rec_id=' . urlencode($rec_id) . '&action=checkoutConfirm' :
+                $urls[0]['url']
```

Comment:
You could support route easily enough, with something like:

`$urls[0]['url'] ?? $this->url($urls[0]['route'] ?? 'home', $urls[0]['routeParams'] ?? [])`

But I don't think we expect a route here, so that's probably going to lead the user to a weird and unexpected place.

Another option might be to leverage the permission denied error page, like this:

`$urls[0]['url'] ?? $this->url('error-permissiondenied', [], ['query' => ['msg' => 'od_docheckout_failure']])`

That should take the user to this:

![image](https://github.com/vufind-org/vufind/assets/309069/15d9a32f-8644-411e-b839-f6d4341217e5)

...which isn't _exactly_ accurate, but is probably close enough to be helpful (by indicating the checkout failed and linking to the support email) if this suspected-impossible situation were to actually arise.

Reply:
I gave the `error-permissiondenied` solution a try and forced that error state.  However I forgot the whole button has a data-lightbox attribute, so what actually happens is this, which I think is unhelpful.

![image](https://github.com/vufind-org/vufind/assets/31278545/a78451bb-f6fb-4954-8249-7ce7aea13f55)

I could fix that with a further check on the presence of `$urls[0]['url']` before adding the lightbox attribute, but I think this has gone to the point where the code becomes confusing and implies valid states that should not actually exist.  So I think actually no change is better, and apologies in advance if this error actually comes up -- would need a smarter solution then based on why it actually happens.

Reply:
Drat, sorry that I didn't think of the lightbox situation. Might it be better to just throw an exception (or, as a softer approach, log an error) if an unexpected state occurs, so it can be tracked down if it does? Or should we just hope for the best? :-)

Reply:
That's a great solution; thank you!